### DISCUSSION TITLE
Multilanguage example

### DISCUSSION THREAD

**Discussion by notflip** [directus-labs/agency-os#13] [Ideas]
Would love to see an example of a block that uses nuxt with i18n, on the frontend as well as the backend, since this is a very common use case! Any idea if this is possible to implement?

**Comment by bryantgillespie** [comment]
> Would love to see an example of a block that uses nuxt with i18n, on the frontend as well as the backend, since this is a very common use case! Any idea if this is possible to implement?

Would be a great example to have for sure. And one we'll explore adding for future iterations of Agency OS.

It's definitely possible to do this - we kept it out of this example because it does add more complexity to both the frontend and the backend. 

On the Directus side, you'd need to create a `translations` collection for each of the different blocks so you can translate the content. Then you'd need to move all the "translatable" fields into the new block translations collection you created for that block. 

You'd also need to change the data fetching on the Nuxt side to get the new schema correctly.

**Reply by notflip** [reply]
That's the point where I'm wondering what the best way to go is, filter the translation on the frontend, so have Directus send all translations, or filter the translations on the backend, so that only the translation for the requested language is returned.
